front/src/components/AdminDataGrid.tsx

```diff
@@ -131,9 +94,82 @@ const rows = [
 ];
 
 
-   export default function DataGridDemo() {
+export default function DataGridDemo() {
+   const router = useRouter();
+
+   // Define the click handler for the "View" button
+   const handleViewClick = (id: number) => {
+     router.push(`/item/${id}`);
```

Comment:
use less javascript == more performance. in the button use this 
`<Button component={Link} href={"/item/"+params.row.id}>`
and import the link component using 
`import Link from 'next/link';`